### ISSUE TITLE
Retrain Taxonomic Model with Pure Taxonomy Data

### ISSUE THREAD

**Issue by marctjones** [marctjones/klareco#443]
## Goal

Retrain the current root embedding model using ONLY taxonomic/hierarchical data sources, removing co-occurrence noise.

## Current Model Issues

The current model (correlation=0.8017, accuracy=0.9027) is **good but mixed**:
- ✅ Good at taxonomic similarity: hund→kok, mus, ĉeval (all animals)
- ⚠️ Noisy from co-occurrence: kat→kronik (0.671) - cat similar to chronicle?
- ⚠️ Mixed signals: Learns BOTH taxonomy AND syntax co-occurrence

**We trained a hybrid model when we wanted a pure taxonomic model.**

## What is Taxonomic Similarity?

**Paradigmatic/IS-A relationships:**
- hundo IS-A mamulo IS-A besto (hierarchical)
- rozo IS-A floro IS-A planto (botanical)
- ruĝa IS-A koloro IS-A kvalito (attribute)

**NOT syntagmatic (co-occurrence):**
- "hundo" and "kuras" co-occur ≠ taxonomically similar
- "manĝas" and "viandon" co-occur ≠ taxonomically similar

### ✅ Use (High Quality Taxonomy)

1. **ReVo hypernym/hyponym relations** (curated)
   - hundo → besto (hypernym)
   - besto → hundo (hyponym)
   - Target: 0.75-0.85 similarity

2. **ReVo part_of/has_part relations**
   - fingro → mano (part-of)
   - mano → fingro (has-part)
   - Target: 0.60-0.70 similarity

3. **Wikipedia category hierarchies** (if available)
   - Extract from Esperanto Wikipedia categories
   - Target: 0.70-0.80 similarity

4. **Fundamento translation overlap** (current - keep)
   - Cross-lingual synonyms from Fundamento
   - Target: 0.70-0.85 similarity

5. **ReVo Jaccard similarity** (current - keep but filter)
   - Only if Jaccard > 0.40 (high overlap)
   - Reduce weight from 2.0 to 1.0
   - Target: 0.40-0.70 similarity

### ❌ Remove (Syntagmatic/Co-occurrence Noise)

1. **Ekzercaro sentence co-occurrence** (REMOVE THIS)
   - Current training: roots within 3-word window
   - This captures syntax, not taxonomy!
   - "La hundo kuras" → (hund, kur) but not taxonomically related

2. **ReVo Jaccard with low overlap** (Jaccard < 0.30)
   - Low overlap may be due to co-occurrence in definitions, not true similarity

## Model Architecture

**Keep current architecture:**
- 64d root embeddings
- MSE loss with graded targets
- Same training loop

**Change only data sources and weights:**
```python
# OLD weights:
Ekzercaro co-occurrence: 10.0  # REMOVE
ReVo Jaccard: 2.0-5.0           # REDUCE to 1.0
ReVo relations: 5.0-8.0         # KEEP
Fundamento: 3.0-5.0             # KEEP

# NEW weights:
ReVo hypernym/hyponym: 10.0     # HIGHEST
ReVo part_of: 8.0               # HIGH
ReVo synonym: 8.0               # HIGH
Fundamento overlap: 5.0         # MEDIUM
ReVo Jaccard (>0.40): 2.0       # LOW
```

## Expected Improvements

1. **Cleaner semantic clusters:**
   - Animals cluster by biological taxonomy, not by "what verbs they co-occur with"
   - Actions cluster by conceptual similarity, not by "what objects they take"

2. **Better generalization:**
   - Unseen animal → placed near semantic neighbors
   - Not biased by corpus frequency

3. **More interpretable:**
   - Similarity = conceptual closeness, not just "used together"

## Validation Tests

Compare OLD model vs NEW model:

| Test | Old Model | Expected New |
|------|-----------|--------------|
| hund → kat | 0.410 | 0.70+ (both animals) |
| kat → kronik | 0.671 | <0.30 (no relation) |
| hund → best | Low? | 0.75+ (hypernym) |
| manĝ → trink | ? | 0.60+ (both consumption verbs) |
| hund → kur | 0.489 | <0.30 (not taxonomically related) |

## Alternative: Keep Both Models?

**YES! This might be better:**
- **Taxonomic Model**: Pure IS-A relationships (this issue)
- **Syntagmatic Model**: Co-occurrence patterns (separate issue #446)
- Use different models for different tasks

Advantages:
- Taxonomic for: category membership, generalization
- Syntagmatic for: query expansion, context prediction
- Explainable: know which type of similarity you're using

## Estimated Effort

- Modify training script to exclude co-occurrence: 1 hour
- Retrain model: 6-8 hours
- Validation: 2 hours
- Compare old vs new: 1 hour
- **Total: 1 day**

## Dependencies

- ✅ ReVo relations file exists
- ✅ Fundamento translations exist
- ✅ Training script exists (modify data loading)

## Deliverables

1. Modified `scripts/train_root_embeddings.py` with `--taxonomy-only` flag
2. New model: `models/root_embeddings/taxonomic_model.pt`
3. Comparison report: old vs new validation results
4. Decision: Replace old model or keep both?

## Next Steps

1. Analyze current model's taxonomy vs syntagmatic performance
2. Modify training script to exclude co-occurrence
3. Retrain with pure taxonomy data
4. Validate and compare
5. Decide: replace or dual-model approach
